packages/proxy/lib/http/util/top-simulation.ts

```diff
@@ -0,0 +1,31 @@
+import { urlOriginsMatch, urlSameSiteMatch } from '@packages/network/lib/cors'
+import type { HttpMiddlewareThis } from '../index'
+import type { SecFetchSite } from '../../types'
+
+export const doesTopNeedToBeSimulated = <T>(ctx: HttpMiddlewareThis<T>): boolean => {
+  const currentAUTUrl = ctx.getAUTUrl()
+
+  // if the AUT url is undefined for whatever reason, return false as we do not want to complicate top simulation
+  if (!currentAUTUrl) {
+    return false
+  }
+
+  // only simulate top if the AUT is NOT the primary origin, meaning that we should treat the AUT as top
+  // or the request is the AUT frame, which is common for redirects and navigations.
+  const doesTopNeedToSimulating = !ctx.remoteStates.isPrimaryOrigin(currentAUTUrl) || ctx.req.isAUTFrame
+
+  return doesTopNeedToSimulating
```

Comment:
if you return directly you can avoid that awkwardly named variable.

```suggestion
return !ctx.remoteStates.isPrimaryOrigin(currentAUTUrl) || ctx.req.isAUTFrame
```

Reply:
fixed in 5a6765c6b6